File: Labo3/src/main/java/be/technifutur/Labo3/model/entities/Utilisateur.java
package be.technifutur.Labo3.model.entities;


import be.technifutur.Labo3.model.entities.enums.Droits;
import lombok.*;
import lombok.experimental.FieldDefaults;

import javax.persistence.*;
import java.awt.*;
import java.util.ArrayList;
import java.util.List;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode
@Entity
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Utilisateur {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;
    @Column(nullable = false,length = 50)
    String name;

    @Column(nullable = false,length = 50)
    String firstName;

    @Column(nullable = false)
    Droits droitAcces;



    @Column(length = 50, nullable = false)
    String pseudo;

    @Column(length = 50,nullable = false)
    String mdp;

    @Embedded
    Adresse adresse;

    @OneToMany(mappedBy = "utilisateur")
    List<Commande> commandes = new ArrayList<>();




}

File: Labo3/angular/LaboFev/src/app/tables/Utilisateur.ts
import { Adresse } from "./Adresse";
import { Commande } from "./Commande";
import { Produit } from "./Produit";


export interface Utilisateur {

    id : number,
    name : String,
    firstName : String,
    droitAcces : String,
    pseudo : String,
    mdp : String,
    adresse : Adresse,
    commandes : Array<Commande>,
    
}